tests/e2e/transaction/transaction_submission.test.ts

```diff
@@ -3,100 +3,362 @@
-          transaction: rawTxn,
+
+        const senderAuthenticator = aptos.signTransaction({ signer: senderAccount, transaction });
+        const secondarySignerAuthenticator = aptos.signTransaction({ signer: secondarySignerAccount, transaction });
+
+        const response = await aptos.submitTransaction({
+          transaction,
+          senderAuthenticator,
+          secondarySignerAuthenticators: { additionalSignersAuthenticators: [secondarySignerAuthenticator] },
+        });
+
+        await waitForTransaction({ aptosConfig: config, txnHash: response.hash });
+      });
+
+      test(
+        "with entry function payload",
+        async () => {
+          const transaction = await aptos.generateTransaction({
+            sender: senderAccount.accountAddress.toString(),
+            secondarySignerAddresses: [secondarySignerAccount.accountAddress.toString()],
+            data: {
+              function: `0x${senderAccount.accountAddress.toStringWithoutPrefix()}::transfer::two_by_two`,
+              arguments: [
+                new U64(100),
+                new U64(200),
+                recieverAccounts[0].accountAddress,
+                recieverAccounts[1].accountAddress,
+                new U64(50),
+              ],
+            },
+          });
+
+          const senderAuthenticator = aptos.signTransaction({ signer: senderAccount, transaction });
+          const secondarySignerAuthenticator = aptos.signTransaction({ signer: secondarySignerAccount, transaction });
+
+          const response = await aptos.submitTransaction({
+            transaction,
+            senderAuthenticator,
+            secondarySignerAuthenticators: { additionalSignersAuthenticators: [secondarySignerAuthenticator] },
+          });
+
+          await waitForTransaction({ aptosConfig: config, txnHash: response.hash });
+        },
+        longTestTimeout,
+      );
+    });
+    describe("fee payer", () => {
+      test("with script payload", async () => {
+        const transaction = await aptos.generateTransaction({
+          sender: senderAccount.accountAddress.toString(),
+          feePayerAddress: feePayerAccount.accountAddress.toString(),
+          data: {
+            bytecode: singleSignerScriptBytecode,
+            arguments: [new U64(1), recieverAccounts[0].accountAddress],
+          },
+        });
+
+        const senderAuthenticator = aptos.signTransaction({ signer: senderAccount, transaction });
+        const feePayerSignerAuthenticator = aptos.signTransaction({ signer: feePayerAccount, transaction });
+
+        const response = await aptos.submitTransaction({
+          transaction,
+          senderAuthenticator,
+          secondarySignerAuthenticators: { feePayerAuthenticator: feePayerSignerAuthenticator },
         });
-        const bobauthenticator = aptos.signTransaction({
-          signer: bob,
-          transaction: rawTxn,
+
+        await waitForTransaction({ aptosConfig: config, txnHash: response.hash });
```

Comment:
I think in general we should be doing a check here right?

The node api will tell you if you have a fee payer  signature, etc

```typescript
const responseSignature = response.signature as TransactionFeePayerSignature;
expect(responseSignature.type).toEqual("fee_payer_signature");
```

this is what I do in mine

the 5 types:

```
ed25519_signature
secp256k1_ecdsa_signature
multi_agent_signature
fee_payer_signature
multi_ed25519_signature
```

Reply:
hmm I guess it depends on what you are testing. those tests tests that transaction is submitted. We have some other tests that tests the signature in `sign_transaction`